src/app/signup/page.tsx

```diff
@@ -1,8 +1,41 @@
+'use client'
+
 import Image from "next/image";
 import { GradButton } from "@/components/ui/grad-button";
 import SignUpImage from "@/components/ui/signup-image";
+import { z } from "zod";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useForm } from "react-hook-form";
 
 const SignUp = () => {
+
+  // Example values for A and B
+  const MIN_NAME_LENGTH = 3; // Minimum length
+  const MAX_NAME_LENGTH = 30; // Maximum length
+
+  const signUpSchema = z.object({
+    fullName: z.string()
+      .min(MIN_NAME_LENGTH, { message: `Name must be at least ${MIN_NAME_LENGTH} characters long` })
+      .max(MAX_NAME_LENGTH, { message: `Name must be no more than ${MAX_NAME_LENGTH} characters long` })
+      .regex(/^[a-zA-Z'-][a-zA-Z' -]*[a-zA-Z'-]$/, { message: "Name can only include alphabetical characters, hyphens, apostrophes, and internal spaces" }),
```

Comment:
This regex seems a bit odd to me as it repeats the pattern 3 times, and the second repetition uses the `*` which should mean "0 or more". Does `/^[a-zA-Z'-]+$/` not work?

Reply:
``/^[a-zA-Z'-]+$/`` this doesn't allow spaces. I am simplifying it to ``/^[a-zA-Z'-]+(?: [a-zA-Z'-]+)*$/``